### ISSUE TITLE
`no-unnecessary-type-assertion` always errors

### ISSUE THREAD

**Issue by noseglid** [AtomLinter/linter-tslint#200]
It seems like this rule (when enabled) always errors for any type assertion. Unnecessary or not:

![screen shot 2017-09-21 at 08 57 23](https://user-images.githubusercontent.com/933880/30682717-e9c58b2e-9eaa-11e7-99d6-3bf7767961ad.png)

This is not a problem when running `tslint` from terminal - which is odd because I can't see that anything special is med on a per-rule basis in this package.

**Comment by Xapphire13** [comment]
Does it occur when you have "use local TSLint" turned on (assuming you have TSLint installed locally)? Or, what version of TSLint are you using on the command line?

**Comment by noseglid** [comment]
I have `Try using the local tslint package (if exists)` enabled.
TSlint on command line:
```
./node_modules/.bin/tslint --version
5.7.0
```

(installed via `yarn add tslint`).

**Comment by Xapphire13** [comment]
The version of TSLint shipped with linter-tslint is currently 5.5.0 if i recall correctly. Could you downgrade to that and run again in command line?

**Comment by noseglid** [comment]
The command line don't complain with `5.5.0`, but this plugin does.

**Comment by noseglid** [comment]
I've done some more digging here. I've verified that:
  * the error actually *do* come from the linter.
  * It is using the correct linter (e.g. the one in my `node_modules`).

I've written a small testscript to run my files with the library rather than the cli:
```javascript
"use strict";
var { Linter, Configuration } = require("tslint");

const program = Linter.createProgram("tsconfig.json");
const files = Linter.getFileNames(program);

const results = files.map(file => {
  console.log("Linting", file);
  const fileContents = program.getSourceFile(file).getFullText();
  const configuration = Configuration.findConfiguration("tslint.json", "./src").results;
  var options = {
      fix: false,
      formatter: "json"
  };
  const linter = new Linter(options, program);
  linter.lint(file, fileContents, configuration);
  return linter.getResult();
});


console.log(results.filter(r => r.errorCount > 0));
```

Which works correctly (e.g. it does _not_ complain about the type assertion). Any idea on how I can continue debug this?

**Comment by Xapphire13** [comment]
Are you saying that the bug **does** reproduce with the script you posted?

> the error actually do come from the linter.

What steps did you take to verify this?

**Comment by noseglid** [comment]
Sorry for being unclear. My script does *not* give the `no-unnecessary-type-assertion`. I did this to rule out the chance that there might be a discrepancy between using the CLI and using the library.

I can see [lintResult](https://github.com/AtomLinter/linter-tslint/blob/master/lib/worker.js#L192) includes the error.

Another thing I noticed is that I get the following error _once_ for each Atom window:
![screen shot 2017-09-22 at 09 45 47](https://user-images.githubusercontent.com/933880/30734319-f4e05bf8-9f7a-11e7-95f2-cd936b687f63.png)

I have `strict` enabled in my `tsconfig.json` which [implies `strictNullChecks`](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

**Comment by Arcanemagus** [comment]
Fixed by #202 (see https://github.com/AtomLinter/linter-tslint/pull/202#issuecomment-331883239).
